File: 卡码/代码随想录/1/08-贪心算法/Go/03-困难/LeetCode_0763.go
// 763. 划分字母区间
package main

import "math"

func partitionLabels(s string) []int {
	hash:=make([]int,30)
	left,right:=0,0
	res:=make([]int,0)
	for i,ch:=range s {
		hash[ch-'a']=i
	}
	for i,ch:=range s {
		right=int(math.Max(float64(right),float64(hash[ch-'a'])))
		if i==right {
			res=append(res,right-left+1)
			left=i+1
		}
	}
	return res
}

File: 卡码/代码随想录/1/08-贪心算法/Js/03-困难/LeetCode_0763.js
//763. 划分字母区间
const partitionLabels = function(s) {
	const hash=new Array(30);
	const res=new Array();
	let left=0,right=0;
	for(let i=0;i<s.length;i++){
		hash[s.charCodeAt(i)-'a'.charCodeAt(0)]=i;
	}
	for(let i=0;i<s.length;i++){
		right=Math.max(right,hash[s.charCodeAt(i)-'a'.charCodeAt(0)]);
		if(i==right){
			res.push(right-left+1);
			left=i+1;
		}
	}
	return res;
};